### PR TITLE
Modularize value-store use

### PR DESCRIPTION
**Pull request by micahhausler**
This will make adding additional backends much easier

### CODE REVIEW

**Review by aelath — APPROVED** [review]
LGTM

File: Makefile [hunk 1]
```diff
@@ -35,7 +35,7 @@ completion: build
 	cp out.sh  /usr/local/etc/bash_completion.d/$(REPO)
 
 docker:
-	docker run --rm -v $$(pwd):/go/src/github.com/skuid/$(REPO) -w /go/src/github.com/skuid/$(REPO) golang:1.8 go build -v -a -tags netgo -installsuffix netgo -ldflags '-w'
+	docker run --rm -v $$(pwd):/go/src/github.com/skuid/$(REPO) -w /go/src/github.com/skuid/$(REPO) golang:1.8-alpine sh -c "apk -U add gcc linux-headers musl-dev && go build -v -a -tags netgo -installsuffix netgo -ldflags '-w'"
```
Comment (ethanfrogers) [inline Makefile:38]: heh, this pattern again.

**Review by ethanfrogers — APPROVED** [review]
LGTM. I like this!

File: cmd/version.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 var Version string
 
 func init() {
-	Version = "v0.2.0"
+	Version = "v0.1.1-dev0"
```
Comment (mrhwick) [inline cmd/version.go:13]: Going backwards?

**Review by mrhwick — APPROVED** [review]
I didn't notice anything apparently wrong in this PR, except for one comment I made about the version number. This LGTM otherwise.

File: cmd/version.go (hunk 2 above)
Comment (micahhausler) [reply cmd/version.go:13]: `v0.2.0` was never released.

**Pull request merged**